### DISCUSSION TITLE
Two computers -> 1 rdio-scanner

### DISCUSSION THREAD

**Discussion by Forts117** [chuot/rdio-scanner#287] [Q&A]
Hi folks,

Is it possible to have two SDRTrunk setups feed into the same rdio-scanner server? Here is my scenario... My home setup is working great, and I'd like to setup a laptop at work to feed another system (that I can't receive at home) into my setup. However I'm not sure this is possible... Any suggestions welcome!

**Comment by chuot** [comment]
The `dirwatch` functionality of Rdio Scanner is only able to monitor local folders, not remote mounted ones. So your option would to setup two rdio-scanners on each PC where sdr-trunk is installed so they can ingest audio files from the local drivey, then use the `downstream` functionnality to propagate the newly ingested audio files to a main instance.

You could also have a single instance on your main PC, and use another instance on your laptop to feed the main instance via a `downstream`.

Since it's so easy to install rdio-scanner, I would recommend to test different scenarios and select the one that suits your needs.

**Reply by Forts117** [reply]
Thanks for the quick reply! I wondered if downstream might be was I was looking for. I'd like to have all the systems available on the phone app if possible, so I'll give this a try.

Thanks very much!

**Reply by chuot** [reply]
Indeed, `downstream` is there for that. Good luck !

**Reply by Forts117** [reply]
Sorry... 1 more question. When setting up the downstream on the remote pc..
- I've created an API at the main PC
- Made a new downstream on the remote PC
- Pasted in the API

The URL I'm confused with. Is it simply https://maininstanceURL:443?

**Reply by Forts117** [reply]
Disregard, got it working! Added /api/call-upload to the URL.

**Reply by Forts117** [reply]
Guess I spoke too soon... Now it doesn't seem to be ingesting any calls. One the remote laptop I see:

downstream: system=1 talkgroup=1234 file=blahblah.mp3 to https://maininstanceURL:443/api/call-upload success

But the calls don't show up...

On the main PC I have the system setup as System 1 and the talkgroups assigned. Not sure what I've done as it was working at one point...

**Reply by chuot** [reply]
What the logs say on the main instance?

**Reply by Forts117** [reply]
They don't show anything coming in at all, just local machine calls being processed.

If the downstream was being blocked on the network somehow, I would assume it wouldn't show success on each call?

**Reply by chuot** [reply]
It shouldn't show success on the originating side (laptop) when downstreaming to the receiving instance if something is wrong on the receiving end (main). If you get success on the laptop side, this means it succeed.

Make sure the ip address defined from the downstream definition on the laptop instance is set with the ip address of the main instance. The api key defined on the downstream (laptop) should be the same as the one defined on the receiving (main) instance in the api keys section.

Your main instance is running on Windows? You should start it at the command line to make sure to get all the logs.

**Comment by Forts117** [comment]
Thanks for getting back... So here is where I'm at..

I generated the API key on the main PC. Copy and pasted this into the downstream API on the laptop
The URL for the main PC has been pasted into the downstream URL box as: https://myaddress.ddns.net:443/api/call-upload  (URL edited for security)
Under Access/Choose systems I have selected 'Everything'

As I mentioned earlier the log on the remote/laptop end shows success for calls uploading, however I'm not seeing anything on the other end. I currently have it running as a service so I'll temporarily suspend that and start it manually instead so I can see what activity is happening.

So does that downstream info look ok, especially the /api/call-upload part? I've not seen any documentation anywhere on how to set this up, I just noticed the URL formatting in Andy Knitt's voxcall software.

On the main machine I've created the same system, with the same talkgroups defined. Then under APIKey I have granted the laptop access to that same system.

Thanks again for your help on this, much appreciated!

**Reply by chuot** [reply]
The url for the downstream definition should be the root url of the main instance, like if you want to access the webapp on the main server.

The /call-upload is a completely different thing, which is normally used for upload-scripts like with trunk-recorder.

So you should have for the downstream url: https://myaddress.ddns.net, nothing more.

You can also test from your laptop if you can access the main instance web app and see if it act normally.

**Reply by Forts117** [reply]
Bingo! That was it! Deleting the /call-upload did the trick.

Thanks so much!

**Reply by chuot** [reply]
Hurray!

Happy Rdio Scanning! Please like and sponsor the open source project if you can.

**Reply by Forts117** [reply]
For sure! I just subscribed to the Android app as well.

Is the csv format for unit IDs available somewhere? It's currently eluding me...

**Reply by chuot** [reply]
Unit ID, Unit Label

**Reply by TPSSCW** [reply]
Hi I'm interested in doing this same thing. I've been running 12 separate feeds to zello and just came across rdio-scanner and think it would beba great solution ! I downloaded rdio-scanner but can't seem to get it all set up, I'm sure it's something simple I'm missing
